src/mscorlib/shared/System/ReadOnlySpan.cs

```diff
@@ -321,5 +321,45 @@ public T[] ToArray()
         /// Returns a 0-length read-only span whose base is the null pointer.
         /// </summary>
         public static ReadOnlySpan<T> Empty => default(ReadOnlySpan<T>);
+
+        /// <summary>Gets an enumerator for this span.</summary>
+        public Enumerator GetEnumerator() => new Enumerator(this);
+
+        /// <summary>Enumerates the elements of a <see cref="ReadOnlySpan{T}"/>.</summary>
+        public ref struct Enumerator
```

Comment:
**Question:** Why does the Enumerator have to be a ref struct here instead of a regular struct?

Reply:
Couldn't have a Span field if it wasn't; also can't implement the interfaces as a ref struct (as they allow boxing to heap)